core/drivers/stm32_gpio.c

```diff
@@ -717,11 +726,29 @@ static int get_pinctrl_from_fdt(const void *fdt, int node,
 			ref->cfg.pupd = pull;
 			ref->cfg.od = odata;
 			ref->cfg.af = alternate;
+			ref->cfg.nsec = pin_non_secure;
+
+			bank_ref = stm32_gpio_get_bank(bank);
+
+			if (pin >= bank_ref->ngpios) {
+				EMSG("node %s requests pin %c%u that does not exist",
+				     fdt_get_name(fdt, consumer_node, NULL),
+				     bank + 'A', pin);
+				do_panic = true;
+			} else if (!pin_is_accessible(bank_ref, pin)) {
+				EMSG("node %s requests pin %c%u which access is denied",
+				     fdt_get_name(fdt, consumer_node, NULL),
+				     bank + 'A', pin);
+				do_panic = true;
```

Comment:
Looking back at this, I really think we shouldn't check if the pinctrl is accessible at this stage but rather only when applying it. For GPIO, I agree, but not for pinctrl. We could boot with a non accessible configuration set by the TDCID coprocessor, then share the peripheral when a service starts